RecoTracker/LST/plugins/LSTPixelSeedInputProducer.cc

```diff
@@ -0,0 +1,223 @@
+    // Get seed tokens
+    const auto& seedToken = seedTokens_[iColl];
+    edm::Handle<edm::View<reco::Track>> seedTracksHandle;
+    iEvent.getByToken(seedToken, seedTracksHandle);
+    const auto& seedTracks = *seedTracksHandle;
+
+    if (seedTracks.empty())
+      continue;
+
+    // Get seed algo
+    edm::EDConsumerBase::Labels labels;
+    labelsForToken(seedToken, labels);
+
+    TString label = labels.module;
+    //format label to match algoName
+    label.ReplaceAll("seedTracks", "");
+    label.ReplaceAll("Seeds", "");
+    label.ReplaceAll("muonSeeded", "muonSeededStep");
+    //for HLT seeds // FIXME: Needed?
+    label.ReplaceAll("FromPixelTracks", "");
+    label.ReplaceAll("PFLowPixel", "");
+    label.ReplaceAll("hltDoubletRecovery", "pixelPairStep");  //random choice
+    int algo = reco::TrackBase::algoByName(label.Data());
+
+    // Get seed track refs
+    edm::RefToBaseVector<reco::Track> seedTrackRefs;
+    for (edm::View<reco::Track>::size_type i = 0; i < seedTracks.size(); ++i) {
+      seedTrackRefs.push_back(seedTracks.refAt(i));
+    }
+
+    edm::ProductID id = seedTracks[0].seedRef().id();
+
+    LogTrace("TrackingNtuple") << "NEW SEED LABEL: " << label << " size: " << seedTracks.size() << " algo=" << algo
+                               << " ProductID " << id;
+
+    for (size_t iSeed = 0; iSeed < seedTrackRefs.size(); ++iSeed) {
+      const auto& seedTrackRef = seedTrackRefs[iSeed];
+      const auto& seedTrack = *seedTrackRef;
+      const auto& seedRef = seedTrack.seedRef();
+      const auto& seed = *seedRef;
+
+
+      if (seedRef.id() != id)
+        throw cms::Exception("LogicError")
+            << "All tracks in 'TracksFromSeeds' collection should point to seeds in the same collection. Now the "
+               "element 0 had ProductID "
+            << id << " while the element " << seedTrackRef.key() << " had " << seedTrackRef.id()
+            << ". The source collection is " << labels.module << ".";
+
+      const bool seedFitOk = !trackFromSeedFitFailed(seedTrack);
+
+      const TrackingRecHit* lastRecHit = &*(seed.recHits().end() - 1);
+      TrajectoryStateOnSurface tsos =
+          trajectoryStateTransform::transientState(seed.startingState(), lastRecHit->surface(), &mf);
+      auto const& stateGlobal = tsos.globalParameters();
+
+      std::vector<int> hitIdx;
+      for (auto const& hit : seed.recHits()) { // FIXME: Is this whole block of code needed?
+        int subid = hit.geographicalId().subdetId();
+        if (subid == (int)PixelSubdetector::PixelBarrel || subid == (int)PixelSubdetector::PixelEndcap) {
+          const BaseTrackerRecHit* bhit = dynamic_cast<const BaseTrackerRecHit*>(&hit);
+          const auto& clusterRef = bhit->firstClusterRef();
+          const auto clusterKey = clusterRef.cluster_pixel().key();
+          hitIdx.push_back(clusterKey);
+        } else if (subid == (int)StripSubdetector::TOB || subid == (int)StripSubdetector::TID ||
```

Comment:
this part does not apply in full, only phase-2 OT is acceptable.
ATM, I'd throw/assert if something else other than pixels is here.